File: app/src/main/java/ssilvalucas/testeandroidv2/screen/home/HomePresenter.java
package ssilvalucas.testeandroidv2.screen.home;

import java.lang.ref.WeakReference;
import ssilvalucas.testeandroidv2.data.model.StatementsResponse;

interface HomePresenterInput{
    void throwError(String errorMessage);
    void onSuccessfulFetchStatements(StatementsResponse response);
}

public class HomePresenter implements HomePresenterInput {
    public WeakReference<HomeActivity> output;

    @Override
    public void throwError(String errorMessage) {
        output.get().showErrorMessage(errorMessage);
    }

    @Override
    public void onSuccessfulFetchStatements(StatementsResponse response) {
        output.get().responseStatements(response);
    }
}

File: app/src/main/java/com/br/natanfelipe/bankapp/presenter/HomePresenter.kt
package com.br.natanfelipe.bankapp.presenter

import com.br.natanfelipe.bankapp.interfaces.home.HomeActivityIntput
import com.br.natanfelipe.bankapp.interfaces.home.HomePresenterInput
import com.br.natanfelipe.bankapp.interfaces.home.LoginActivityIntput
import com.br.natanfelipe.bankapp.view.home.HomeResponse
import com.br.natanfelipe.bankapp.view.home.HomeViewModel
import java.lang.ref.WeakReference

class HomePresenter : HomePresenterInput {

    lateinit var output : WeakReference<HomeActivityIntput>

    override fun presentHomeMetaData(response: HomeResponse) {
        var homeViewModel = HomeViewModel()
        homeViewModel.billsList = arrayListOf()

        if (!response.billsList.isNullOrEmpty()) {
            for (b in response.billsList) {
                homeViewModel.billsList.add(b)
            }
            output.get()?.displayHomeMetaData(homeViewModel)
        }
    }


}